magic_marker/magic_marker.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+
+"""MagicMarker is a tool that automatically marks pytests with a UUID"""
+# ======================================================================================================================
+# Imports
+# ======================================================================================================================
+from __future__ import absolute_import
+from flake8.main.cli import main as flake8_main
+from contextlib import contextmanager
+import sys
+import six
+import json
+import uuid
+import re
+import tempfile
+import string
+import random
+import shutil
+import errno
+import os
+
+
+class MagicMarker(object):
+
+    def __init__(self, mark):
+        """Crate a new MagicMarker object
+
+        Args:
+            mark (str): The mark to create
+        """
+        self._mark = mark
+        dir_name = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10))
+        self._backup_dir = os.path.join(tempfile.gettempdir(), dir_name)
+
+    @property
+    def backup_dir(self):
+        return self._backup_dir
+
+    def run_flake8_and_mark(self, directory):
+        """Run flak8 and edit and fix errors
+
+        Args:
+            directory (str): The directory to target for the fix
+
+        Returns:
+            Str: the message stating what was performed
+        """
+
+        args = [
+            'flake8',
+            directory,
+            "--format=json",
+            "--pytest-mark1=name={},value_match=uuid".format(self._mark),
+            "--select=M501"  # only covers case of no mark present
+        ]
+
+        with self.patch_sys_argv(args), self.captured_stdout() as stdout:
+            try:
+                flake8_main()
+            except SystemExit:
+                pass  # This is raised by flake8
+
+        flake8_output = json.loads(stdout.getvalue())
+        self._backup_whole_dir(directory)
```

Comment:
What happens if the user passes in a path to file instead of a directory?

Reply:
For now the file is named with the random string.  Arguably not the best